pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -694,6 +694,10 @@ type KubeProxyConfig struct {
 	// defaults to IPTables.
 	// +optional
 	Mode *ProxyMode `json:"mode,omitempty" protobuf:"bytes,2,opt,name=mode,casttype=ProxyMode"`
+	// Enabled indicates whether kube-proxy should be deployed or not.
+	// defaults to true if not specified.
```

Comment:
Maybe you can add to these doc strings that it's depending on the used network extension whether it's possible/feasible to disable it

Reply:
Updated documentation comment with new commit.
As it is duplicated multiple times here is the diff:
`+       // Depending on your networking provider extensions switching kube-proxy off might be rejected.`
`+       // For example, cilium supports running without kube-proxy, but calico usually needs kube-proxy except for its ebpf dataplane.`